packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -64,15 +66,153 @@ export function getIdentUrl(ident: Ident) {
   }
 }
 
-export async function get(path: string, {configuration, headers, ident, authType, registry, ...rest}: Options) {
-  if (ident && typeof registry === `undefined`)
-    registry = npmConfigUtils.getScopeRegistry(ident.scope, {configuration});
+export type GetPackageMetadataOptions = Omit<Options, 'ident'> & {
+  /**
+   * Warning: This option will return all cached metadata if the version is found, but the rest of the metadata can be stale.
+   */
+  version?: string;
+};
+
+// We use 2 different caches:
+// - an in-memory cache, to avoid hitting the disk and the network more than once per process for each package
+// - an on-disk cache, for exact version matches and to avoid refetching the metadata if the resource hasn't changed on the server
+
+const PACKAGE_METADATA_CACHE = new Map<IdentHash, PackageMetadata>();
```

Comment:
It nicely mirrors the [singleton `cache` in `httpUtils`](https://github.com/yarnpkg/berry/blob/ad03119d78b54208b0748a70e537727ef7fdd7ba/packages/yarnpkg-core/sources/httpUtils.ts#L18), which it supercedes (as that one is disabled when `wrapNetworkRequest` is used since it can differ between calls).